File: single-language-analysis/dataset-files/python/test_smell_102.py
# Original URL: https://github.com/Phlogistique/qibuild/blob/27876e99deac05f60984468ec4eb48ba693dbc2c/python/qipkg/test/test_qipkg.py#L71-L82

def test_smell_102(qipkg_action):
    dump_syms = qisys.command.find_program("dump_syms", raises=False)
    if not dump_syms:
        return

    a_cpp_proj = qipkg_action.add_test_project("a_cpp")
    pml = os.path.join(a_cpp_proj.path, "a_cpp.pml")

    qipkg_action("configure", "--release", "--with-debug-info", pml)
    qipkg_action("build", pml)
    pkg, symbols_archive = qipkg_action("make-package", "--with-breakpad", pml)
    assert os.path.exists(symbols_archive)

File: single-language-analysis/dataset-files/java/TestSmell102.java
import org.junit.Test;
import static org.junit.Assert.*;
import java.io.File;

public class TestSmell102 {

    @Test
    public void testBreakpadSymbols() {
        String dumpSyms = qisys.command.findProgram("dump_syms", false);
        if (dumpSyms == null) {
            return;
        }

        QipkgAction aCppProj = qipkgAction.addTestProject("a_cpp");
        String pml = aCppProj.path + "/a_cpp.pml";

        qipkgAction("configure", "--release", "--with-debug-info", pml);
        qipkgAction("build", pml);
        Object[] result = qipkgAction("make-package", "--with-breakpad", pml);
        Object pkg = result[0];
        Object symbolsArchive = result[1];
        assertTrue(new File((String) symbolsArchive).exists());
    }
}